core/src/org/javarosa/xml/TreeElementParser.java

```diff
@@ -55,6 +56,9 @@ public TreeElement parse() throws InvalidStructureException, IOException,
                 case KXmlParser.TEXT:
                     element.setValue(new UncastData(parser.getText().trim()));
                     break;
+                case KXmlParser.END_DOCUMENT:
+                    // top-level escape, when the parser has reached the end of the document
+                    return element;
```

Comment:
Needed to exit the loop, since a root tag(s) are wrapped in a top level null-named TreeElement